### PR TITLE
Sphinx

### PR DESCRIPTION
**Pull request by BibMartin**
Update sphinx documentation.

**Not finished** : we need to move the docstrings of `__init__` methods to the class docstring.

For example:

```python
class Foo(object):
    __init__(self):
        """A docstring"""
        pass
    some_method(self):
        """Method docstring"""
```

Has to be trandformed into

```python
class Foo(object):
    """A docstring"""
    __init__(self):
        pass
    some_method(self):
        """Method docstring"""
```

### CODE REVIEW

**Comment by BibMartin** [comment]
To build the docs:

    pip install sphinx
    cd ~/sandbox_path/folium
    rm -rf docs/_build/
    sphinx-build -b html docs docs/_build/ 

Then open `~/sandbox_path/folium/docs/_build/index.html`.

**Comment by BibMartin** [comment]
> Showing  54 changed files  with 1,206 additions and 5,185 deletions.

Just to beat @ocefpaf's tweet :wink: and no functionality at all.

**Comment by ocefpaf** [comment]
> Just to beat @ocefpaf's tweet :wink: and no functionality at all.

That is a 4.29 ratio!  Nice!!

(Oh, and nice PR too BTW.)

**Comment by BibMartin** [comment]
I'd say this PR is finished.

There are still a lot of `TODO: docstring` in the code, but let's do them in another PR (to separate content from machine-plugging).

File: docs/conf.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 
 # Add any Sphinx extension module names here, as strings. They can be extensions
 # coming with Sphinx (named 'sphinx.ext.*') or your custom ones.
-extensions = ['sphinx.ext.autodoc']
+extensions = ['sphinx.ext.autodoc', 'sphinxcontrib.napoleon']
```
Comment (ocefpaf) [inline docs/conf.py]: No need for `sphinxcontrib.napoleon`  just use `sphinx.ext.napoleon`

**Comment by ocefpaf** [comment]
Need to update folium version in https://github.com/python-visualization/folium/pull/334/files#diff-85987f48f1258d9ee486e3191495582dR51

File: docs/installing.rst [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+$ py.test
+
+Build the docs
+--------------
+
+To build the docs, you'll also need to install:
+::
+
+ sphinx
+ sphinxcontrib-napoleon
```
Comment (ocefpaf) [inline docs/installing.rst]: You can remove this extension from the instructions.  (`napoleon` is not part of sphinx.)

**Comment by BibMartin** [comment]
@ocefpaf 
> You can remove this extension from the instructions. (napoleon is not part of sphinx.)

> Need to update folium version in https://github.com/python-visualization/folium/pull/334/files#diff-85987f48f1258d9ee486e3191495582dR51

Done.

I've started to fill `quickstart.rst` btw.

To visualize the docs (without building it), you can use

http://journois.pythonanywhere.com/static/folium_docs/index.html

I'll try to keep it up-to-date.

**Comment by ocefpaf** [comment]
> Done.

Can I merge it?

> I've started to fill quickstart.rst btw.

Nice!

> To visualize the docs (without building it), you can use
> http://journois.pythonanywhere.com/static/folium_docs/index.html
> I'll try to keep it up-to-date.

Cool. I guess we can just push that to `gh-pages`, might be easier.

**Comment by BibMartin** [comment]
> Can I merge it?

Yes if you're okay.

> Cool. I guess we can just push that to gh-pages, might be easier.

It does not work with github pages (see http://bibmartin.github.io/folium_docs/) because there are internal links into the generated html code. I think we would need to push the `rst` output on `gh-pages` if we wished to have it work.

**Comment by ocefpaf** [comment]
> It does not work with github pages (see http://bibmartin.github.io/folium_docs/) because there are internal links into the generated html code. I think we would need to push the `rst` output on `gh-pages` if we wished to have it work.

I think there is a way, but I usually do this once and forget about it.  I always have to re-learn everything.  Anyways, http://journois.pythonanywhere.com/static/folium_docs/index.html is cool for a quick viz.   We'll figure something out later on.

BTW: We have docs!

yes

**Comment by BibMartin** [comment]
@ocefpaf
Thanks for review and merge.

> I think there is a way, but I usually do this once and forget about it. I always have to re-learn everything. 

So do I. I think also that it's possible ; but cannot figure out how (for the moment :wink: )

> Anyways, http://journois.pythonanywhere.com/static/folium_docs/index.html is cool for a quick viz.  We'll figure something out later on.

Well, pythonanywhere is okay for a demo, but we cannot host the docs there.
I would prefer to update the docs in *readthedocs*. 
Unless you know how to do, It would be helpful if @wrobstory told us how to do that (and give some rights if needed). (please)

> BTW: We have docs!

Yes an no ; there are many TODO to fix before closing #140. I'll try to tackle that soon to have the release in January.

**Comment by ocefpaf** [comment]
> So do I. I think also that it's possible ; but cannot figure out how (for the moment :wink: )

I guess that

```shell
make -C docs html
ghp-import -n -p docs/_build/html 
```

from the root directory works.  (check http://ocefpaf.github.io/folium/)

> I would prefer to update the docs in readthedocs.

:+1:  I will look into that.

**Comment by BibMartin** [comment]
> (check http://ocefpaf.github.io/folium/)

Really nice !
If it's that simple maybe we don't nedd *readthedocs* anymore, do we ?

**Pull request merged**